File: vuln-sandbox/ruby/cwe-78/command-injection.rb
# CWE-78: OS Command Injection via system() in Ruby
# Rule: ruby-command-injection-system
# Expected: TruePositive

class FilesController < ApplicationController
  def convert
    # VULNERABLE: User input passed directly to system()
    filename = params[:filename]
    system("convert #{filename} output.png")
    render plain: "Converted"
  end

  def resize
    # VULNERABLE: User input in exec()
    exec("mogrify -resize #{params[:size]} #{params[:file]}")
  end
end

File: vuln-sandbox/php/cwe-78/command-injection.php
<?php
// CWE-78: OS Command Injection via exec() in PHP
// Rule: php-command-injection-exec
// Expected: TruePositive

// VULNERABLE: User input passed directly to exec()
$filename = $_GET['filename'];
exec("convert $filename output.png");

// VULNERABLE: User input in system()
$size = $_POST['size'];
system("mogrify -resize $size image.jpg");

// VULNERABLE: User input in shell_exec()
$cmd = $_REQUEST['cmd'];
$output = shell_exec("ls $cmd");
echo $output;
?>
